src/net/parsers/texttracks/webvtt.js

```diff
@@ -0,0 +1,311 @@
+/** 
+* Parse WebVTT from text. Returns an array with:
+* - start : start of current cue, in seconds
+* - end : end of current cue, in seconds
+* - content : HTML formatted cue.
+*
+* Global style is parsed and applied to div element. 
+* Specific style is parsed and applied to class element. 
+*
+* @param {string} text
+* @return {Array<Object>}
+*/
+export default function parseWebVTT(text) {
+  const newLineChar = /\n|\s{2,}/g;
+  const linified = text.split(newLineChar);
+  const cuesArray = [];
+  const styleElements = [];
+
+  if (!linified[0].match(/^WEBVTT.*/)) {
+    console.log("Can't parse WebVTT: Non valid file.");
+    return;
+  }
+
+  let i = 0;
+
+  // Find every style element into WebVTT, and collect style 
+  // associated with class / tags
+  for (i = 1; i<linified.length; i++) {
+    if (!(linified[i].match(/^\r/) || linified[i].length === 0)) {
+      if(findElementType(linified[i]) === "style"){
+        parseStyle(i+1, linified, styleElements);        
+      }
+      while (linified[i] 
+        && !(linified[i].match(/^\r/) || linified[i].length === 0)) {
+        i++;
+      }
+    }
+  }
+
+  // Parse cues, format and apply style.
+  for (i = 1; i < linified.length; i++) {
+    if (!(linified[i].match(/^\r/) || linified[i].length === 0)) {
+      if (findElementType(linified[i]) === "cue") {
+        cuesArray.push(parseCue(i, linified, styleElements));
+      }
+      while (linified[i] 
+        && !(linified[i].match(/^\r/) || linified[i].length === 0)) {
+        i++;
+      }
+    }
```

Comment:
Why not calculating the end of the style before parsing it?
That way we could just do at the end:
```js
const styleBlock = linified.slice(startingI, endingI);
const parsedStyles = parseStyleBlock(styleBlock);
styleElements.push(...parsedStyles);
```
Which just reads itself and is easier to follow

Reply:
Overall, I prefer something like:
```js
for (let i = 1; i < linified.length; i++) {
    if (isStartOfStyleBlock(linified[i])) {
      const startOfStyleBlock = i;
      i++;

      // continue incrementing i until either:
      //   - empty line
      //   - end of file
      while(linified[i]) {
        i++;
      }
      const styleBlock = linified.slice(startOfStyleBlock, i);
      const parsedStyles = parseStyleBlock(styleBlock);
      styleElements.push(...parsedStyles);
   }
}
```
It's easier to read.